### PR TITLE
perf(neon_framework): Store cached requests as BLOB

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/nextcloud/neon/issues/2160

I didn't do any performance comparisons (maybe @Leptopoda you can do that on one of your aging devices?), but at least the image flickering seems to be less which would indicate the cache is faster for binary data.

### CODE REVIEW

**Review by Leptopoda — COMMENTED** [review]
I'll do a performance check shortly

**Comment by provokateurin** [comment]
The dynamite_runtime commit is not breaking because it only widens the type.

File: packages/neon_framework/lib/src/utils/request_manager.dart [hunk 1]
```diff
@@ -92,13 +92,13 @@ class RequestManager {
         },
         unwrap: (rawResponse) => unwrap(rawResponse),
         serialize: (data) {
-          const encoder = RawResponseEncoder();
-
-          return json.encode(encoder.convert(data as DynamiteRawResponse));
+          return const RawResponseEncoder()
+              .fuse(const JsonEncoder())
+              .fuse(const Utf8Encoder())
```
Comment (Leptopoda) [inline packages/neon_framework/lib/src/utils/request_manager.dart:97]: Any reason you did not use the mentioned `JsonUtf8Encoder`?

**Review by Leptopoda — APPROVED** [review]
Looks like the performance has improved quite a bit.

File: packages/neon_framework/lib/src/utils/request_manager.dart (hunk 1 above)
Comment (provokateurin) [reply packages/neon_framework/lib/src/utils/request_manager.dart:97]: Only to make it more similar to the deserialization part. If you want I can change it, but I don't know if it makes any difference in the end.

Comment (provokateurin) [reply packages/neon_framework/lib/src/utils/request_manager.dart:97]: Ok from the class description it sounds like it needs one less copy of the data and thus should be more performant.

**Pull request merged**